pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -200,10 +200,9 @@ func remediate(r *ReconcileMachineHealthCheck, remediationStrategy healthcheckin
 		// apply remediation logic, if at least one condition last more than specified timeout
 		// specific remediation logic goes here
 		if unhealthyForTooLong(nodeCondition, conditionTimeout) {
-			switch remediationStrategy {
-			case remediationStrategyReboot:
+			if remediationStrategy != nil && *remediationStrategy == remediationStrategyReboot {
 				return r.remediationStrategyReboot(machine, node)
-			default:
+			} else {
```

Comment:
Nit: no need for the `else` part given `return` in the `if` branch. Though, given the list of strategies might grow over time, it's just a nit.

Reply:
I agree. I'd actually prefer this remained a `swtich`, but it's not a big deal. We can change this as the list of remediation strategies grows.